--- a/ipq806x/nss_gmac_ctrl.c
+++ b/ipq806x/nss_gmac_ctrl.c
@@ -545,7 +545,7 @@ static int nss_gmac_cadj(struct device *
 static int nss_gmac_fadj(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
 {
 	struct nss_gmac_dev *gmacdev = (struct nss_gmac_dev *)netdev_priv(to_net_dev(dev));
-	int64_t offset = 0;
+	uint64_t offset = 0;
 	uint64_t direction = 0;
 	uint32_t sec;
 	uint32_t nsec;
