--- a/source3/auth/auth.c
+++ b/source3/auth/auth.c
@@ -526,12 +526,12 @@ NTSTATUS make_auth3_context_for_ntlm(TAL
 	switch (lp_server_role()) {
 	case ROLE_DOMAIN_MEMBER:
 		DEBUG(5,("Making default auth method list for server role = 'domain member'\n"));
-		methods = "guest sam winbind:ntdomain sam_ignoredomain";
+		methods = "guest sam sam_ignoredomain";
 		break;
 	case ROLE_DOMAIN_BDC:
 	case ROLE_DOMAIN_PDC:
 		DEBUG(5,("Making default auth method list for DC\n"));
-		methods = "guest sam winbind:trustdomain sam_ignoredomain";
+		methods = "guest sam sam_ignoredomain";
 		break;
 	case ROLE_STANDALONE:
 		DEBUG(5,("Making default auth method list for server role = 'standalone server', encrypt passwords = yes\n"));
